feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,271 @@
+        variables.
+        """
+
+        return MeanEncoder(
+            variables=self.variables_categorical_, errors="raise"
+        )
+
+    def _make_combined_pipeline(self):
+
+        encoder_num = MeanEncoder(variables=self.variables_numerical_, errors="raise")
+        encoder_cat = MeanEncoder(variables=self.variables_categorical_, errors="raise")
+
+        _pipeline_combined = Pipeline(
+            [
+                ("discretisation", self._make_discretiser()),
+                ("encoder_num", encoder_num),
+                ("encoder_cat", encoder_cat),
+            ]
+        )
+
+        return _pipeline_combined
+
+    def _make_discretiser(self):
+        """
+        Instantiate either EqualWidthDiscretiser or EqualFrequencyDiscretiser.
+        """
+        if self.strategy == "equal-width":
+            discretiser = EqualWidthDiscretiser(
+                bins=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+        else:
+            discretiser = EqualFrequencyDiscretiser(
+                q=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+
+        return discretiser
+
+    def r_squared_score(self, X: pd.DataFrame, y: pd.Series,
+                        sample_weight: float = None) -> float:
+        """
+        Returns the coefficient of determination, which represents the proportion of
+        the variation in 'y' that can be predicted by 'X'.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            Test samples.
+
+        y : pandas series of shape = [n_samples, ]
+            True lables for 'X'.
+
+        sample_weight : pandas series of shape [n_samples, ], default=None
+            Sample weights.
+
+        Returns
+        --------
+        r2_score : float
+            Linear correlation between 'y' and 'y_pred'.
+
+        """
+        # Should we use the same or different nomenclature as sklearn?
+
+        y_pred = self.predict(X)
+        return r2_score(y, y_pred, sample_weight=sample_weight)
+
+    def mean_accuracy_score(self, X: pd.DataFrame, y: pd.Series,
+                            sample_weight: float = None) -> float:
+        """
+        Derives the mean accuracy of the given test data and labels.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            Test samples.
+
+        y : pandas series of shape = [n_samples, ]
+
+        sample_weight : pandas series of shape [n_samples, ], default=None
+            Sample weights.
+
+        Returns
+        --------
+        accuracy_score : float
+            Mean accuracy of 'self.predict(X)' and 'y'.
+
+        """
+        # Should we use the same or different nomenclature as sklearn?
+
+        y_pred = self.predict(X)
+        return accuracy_score(y, y_pred, sample_weight=sample_weight)
```

Comment:
I know that you copied sample_weight=sample_weight from the sklearn code. But I think we need to default to None here.

Reply:
`score()` is defined by the following code:

`def score(self, X: pd.DataFrame, y: pd.Series, regression: bool = True,
              sample_weight: float = None) -> float:`

Isn't `sample_weight = None` the default?